instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorE2ETests.kt

```diff
@@ -0,0 +1,481 @@
+     */
+    @Test
+    fun rum_rummonitor_add_custom_action_with_no_outcome() {
+        val testMethodName = "rum_rummonitor_add_custom_action_with_no_outcome"
+        val viewKey = forge.aViewKey()
+        val viewName = forge.aViewName()
+        val actionName = forge.anActionName()
+        executeInsideView(viewKey, viewName, testMethodName) {
+            measure(testMethodName) {
+                GlobalRum.get().addUserAction(
+                    RumActionType.CUSTOM,
+                    actionName,
+                    attributes = defaultTestAttributes(testMethodName)
+                )
+            }
+        }
+    }
+
+    /**
+     * apiMethodSignature: RumMonitor#fun addUserAction(RumActionType, String, Map<String, Any?>)
+     */
+    @Test
+    fun rum_rummonitor_add_action_with_outcome() {
+        val testMethodName = "rum_rummonitor_add_action_with_outcome"
+        val viewKey = forge.aViewKey()
+        val viewName = forge.aViewName()
+        val actionName = forge.anActionName()
+        executeInsideView(viewKey, viewName, testMethodName) {
+            measure(testMethodName) {
+                GlobalRum.get().addUserAction(
+                    forge.aValueFrom(RumActionType::class.java),
+                    actionName,
+                    attributes = defaultTestAttributes(testMethodName)
+                )
+            }
+            sendRandomActionOutcomeEvent(forge)
+        }
+    }
+
+    /**
+     * apiMethodSignature: RumMonitor#fun addUserAction(RumActionType, String, Map<String, Any?>)
+     */
+    @Test
+    fun rum_rummonitor_add_custom_action_while_active_action() {
+        val testMethodName = "rum_rummonitor_add_custom_action_while_active_action"
+        val viewKey = forge.aViewKey()
+        val viewName = forge.aViewName()
+        val activeActionName = forge.anActionName(prefix = "rumActiveAction")
+        val customActionName = forge.anActionName()
+        val type = forge.aValueFrom(RumActionType::class.java)
+        executeInsideView(viewKey, viewName, testMethodName) {
+            GlobalRum.get().startUserAction(
+                type,
+                activeActionName,
+                attributes = defaultTestAttributes(testMethodName)
+            )
+            measure(testMethodName) {
+                GlobalRum.get().addUserAction(
+                    RumActionType.CUSTOM,
+                    customActionName,
+                    attributes = defaultTestAttributes(testMethodName)
+                )
+            }
+            sendRandomActionOutcomeEvent(forge)
+            GlobalRum.get().stopUserAction(type, activeActionName, forge.exhaustiveAttributes())
+        }
+    }
+
+    // endregion
+
+    // region Resource
+
+    /**
+     * apiMethodSignature: RumMonitor#fun startResource(String, String, String, Map<String, Any?> = emptyMap())
+     */
+    @Test
+    fun rum_rummonitor_start_resource() {
```

Comment:
I am missing 2 test cases here where a view could not be stopped if it is waiting for a pending resource or action